app/views/split_checkout/_your_details.html.haml

```diff
@@ -90,25 +90,23 @@
                 = t(:checkout_method_free)
 
 
-    - if true # todo: Checkout.requireShipAddress()  //   requireShipAddress: -> @shippingMethod()?.require_ship_address
-      %div.checkout-input
-        = f.check_box "Checkout.ship_address_same_as_billing", { id: "Checkout.ship_address_same_as_billing" } # "ng-model": "Checkout.ship_address_same_as_billing"}s
-        = f.label "Checkout.ship_address_same_as_billing", t(:checkout_address_same), { for: "Checkout.ship_address_same_as_billing" }
-        
-    - if spree_current_user || true # todo: && Checkout.requireShipAddress()  //   requireShipAddress: -> @shippingMethod()?.require_ship_address
-      %div.checkout-input
+    %div.checkout-input{"data-splitcheckout-target": "requireShipAdressInput"}
+      = f.check_box "Checkout.ship_address_same_as_billing", { id: "Checkout.ship_address_same_as_billing" } # "ng-model": "Checkout.ship_address_same_as_billing"}s
+      = f.label "Checkout.ship_address_same_as_billing", t(:checkout_address_same), { for: "Checkout.ship_address_same_as_billing" }
+      
+    - if spree_current_user
+      %div.checkout-input{"data-splitcheckout-target": "requireShipAdressInput"}
         = f.check_box "Checkout.default_ship_address", { id: "Checkout.default_ship_address" }
         = f.label "Checkout.default_ship_address", t(:checkout_default_ship_address), { for: "Checkout.default_ship_address" }
 
-    - if true # todo: Checkout.shippingMethod().description
-      .div.checkout-input
-        #distributor_address.panel
-          %span{}{{ Checkout.shippingMethod().description }}
-          %br/
-          %br/
-          - if @order.order_cycle.pickup_time_for(@order.distributor)
-            = t :checkout_ready_for
-          = @order.order_cycle.pickup_time_for(@order.distributor)
+    %div.checkout-input{"data-splitcheckout-target": "shippingMethodDescription"}
```

Comment:
Yay! :heart:

In terms of designing Stimulus controllers, there's a slightly different we can potentially approach them that's a bit different to component-based javascript. So instead of looking at them in terms of their context and the elements they're attached to, we can look at them in a generic way that's based on what behavior they are adding. So for example here, the behavior is that you can click on some elements and it will show/hide some associated description text, right? But in theory that bit of behavior isn't unique to this shipping methods form... the exact same behavior could be used on the payment methods form, or on a totally different form, or a set of clickable tabs that show some info.

If we shift the conceptual focus a bit from being a "checkout page shipping methods descriptions controller" to a "display descriptions controller" then the same controller might be used in lots of pages instead of just one.

There's a nice primer on it here if you're interested: https://boringrails.com/articles/better-stimulus-controllers/

Reply:
But maybe we can start with this for now and refactor it later if it makes sense :+1:

Reply:
I tested, and tried to create a toggle controller: 
bdf0539

https://github.com/jibees/openfoodnetwork/blob/bdf0539a7a816caf10023c39a9924c4a5d9e5e85/app/webpacker/controllers/toggle_controller.ts#L1


Usage: 

```haml
%div{ "data-controller": "toggle" }
  %a{ "data-action": "toggle#toggle", "data-toggle-show":  condition_to_display_the_targets ? true || false }

%div{ "data-toggle-target": "content", style: "display: none" }
%div{ "data-toggle-target": "content", style: "display: none" }
```

I wanted to create a small unit test, but I miserably failed.